src/GafferRenderMan/RenderManShader.cpp

```diff
@@ -372,24 +360,278 @@ Gaffer::Plug *loadParameter( const boost::property_tree::ptree &parameter, Plug
+						boost::lexical_cast<float>( valueTokens[3*i] ),
+						boost::lexical_cast<float>( valueTokens[3*i + 1] ),
+						boost::lexical_cast<float>( valueTokens[3*i + 2 ] )
+					)
+				}
+			);
+		}
+
+		defaultValue.trimEndPoints();
+
+		return new SplinefColor3fPlug( baseName, direction, defaultValue , Plug::Default );
+	}
+	else
+	{
+		SplineDefinitionff defaultValue;
+		defaultValue.interpolation = interpolationDefault;
+
+		for( unsigned int i = 0; i < positionsDefault.size(); i++ )
+		{
+			defaultValue.points.insert(
+				std::pair{ positionsDefault[i], boost::lexical_cast<float>( valueTokens[i] ) }
+			);
+		}
+
+		defaultValue.trimEndPoints();
+
+		return new SplineffPlug( baseName, direction, defaultValue , Plug::Default );
+	}
+}
+
+
 void loadParameters( const boost::property_tree::ptree &tree, Plug *parent, const ParameterSet *omit, std::unordered_set<const Plug *> &validPlugs )
 {
+	// In order to assemble together the parameters needed to build a spline, we need to be able to look up
+	// parameters by name. Build a map of all the parameters that we should be building plugs from
+	std::map<std::string, const boost::property_tree::ptree *> parameters;
+
 	for( const auto &child : tree )
 	{
-		if( child.first == "param" )
+		// Recurse for sub categories
+		if( child.first == "page" )
 		{
-			if( omit && omit->count( child.second.get<string>( "<xmlattr>.name" ) ) )
+			loadParameters( child.second, parent, omit, validPlugs );
+		}
+		else if( child.first == "param" )
+		{
+			const string name = child.second.get<string>( "<xmlattr>.name" );
+			if( omit && omit->count( name ) )
 			{
 				continue;
 			}
-			if( Plug *p = loadParameter( child.second, parent ) )
+
+			parameters[name] = &child.second;
+		}
+	}
+
+
+	std::vector<string> removeParameters;
+
+	for( const auto &param : parameters )
+	{
+		PlugPtr spline;
+		try
+		{
+			spline = findSplinePlugFromPositionsParameter( param.first, *param.second, parameters, parent->direction(), removeParameters );
+		}
+		catch( std::exception &e )
+		{
+			msg(
+				IECore::Msg::Warning, "RenderManShader::loadShader",
+				fmt::format( "Error while parsing spline based on \"{}\" : {}", param.first, e.what() )
+			);
+		}
+
+		if( spline )
+		{
+			parent->addChild( spline );
+			validPlugs.insert( spline.get() );
+		}
+	}
+
+	// Remove any parameters that have already been turned into SplinePlugs
+	for( const std::string &i : removeParameters )
+	{
+		parameters.erase( i );
+	}
+
+	// Load the basic parameters
+	for( const auto &param : parameters )
+	{
+		try
+		{
+			if( Plug *p = loadParameter( *param.second, parent ) )
 			{
 				validPlugs.insert( p );
 			}
 		}
-		else if( child.first == "page" )
+		catch( std::exception &e )
 		{
-			loadParameters( child.second, parent, omit, validPlugs );
+			msg(
+				IECore::Msg::Warning, "RenderManShader::loadShader",
+				fmt::format( "Error while parsing parameter \"{}\" : {}", param.first, e.what() )
+			);
+			continue;
```

Comment:
This breaks the natural ordering of the parameters as they're specified in the `.args` file. See below, where the mode plug is now in the middle rather than the top :

<img width="501" height="150" alt="image" src="https://github.com/user-attachments/assets/c10617a9-79c0-4525-a057-06b7dfff4173" />
 
As well as getting the right ordering for the basic parameters, we need to make sure that the spline parameters are interleaved with them correctly.